docs/TypedMongo.md

```diff
@@ -0,0 +1,446 @@
+* Single-field order, e.g. `MyEntity.ref(_.int).ascending`
+* Multi-field order, e.g. `MongoDocumentOrder(MyEntity.ref(_.int) -> true, MyEntity.ref(_.data.number) -> false)`
+
+For more examples, see the Scaladoc or
+[tests](https://github.com/AVSystem/scala-commons/blob/mongo-api/commons-mongo/src/test/scala/com/avsystem/commons/mongo/typed/MongoOrderTest.scala).
+
+### `MongoDocumentUpdate`
+
+A `MongoDocumentUpdate[E]` represents a MongoDB update document for an entity of type `E`. Example:
+
+```scala
+val update: MongoDocumentUpdate[MyEntity] = 
+  MyEntity.ref(_.strOpt).set(Opt("foo")) && MyEntity.ref(_.int).inc(5)
+```
+
+For more examples, see the Scaladoc or
+[tests](https://github.com/AVSystem/scala-commons/blob/mongo-api/commons-mongo/src/test/scala/com/avsystem/commons/mongo/typed/MongoUpdateTest.scala).
+
+### `MongoIndex`
+
+A `MongoIndex[E]` represents a MongoDB index document for an entity of type `E`. Examples:
+
+* Single field index, e.g. `MyEntity.ref(_.int).ascendingIndex`
+* Multi-field index, e.g.
+
+  ```scala
+  import MongoIndexType._
+  val index: MongoIndex[MyEntity] =
+  MongoIndex(MyEntity.ref(_.str) -> Hashed, MyEntity.ref(_.int) -> Descending)
+  ```
+  
+For more examples, see the Scaladoc or
+[tests](https://github.com/AVSystem/scala-commons/blob/mongo-api/commons-mongo/src/test/scala/com/avsystem/commons/mongo/typed/MongoIndexTest.scala).
+
+## Invoking database commands
+
+In order to invoke database commands, create a `TypedMongoCollection` which is a wrapper over Reactive Streams
+driver's raw `MongoCollection`. 
+
+(assuming sample entity classes from the [previous section](#core-types))
+
+```scala
+import com.mongodb.reactivestreams.client.MongoClients
+
+val client = MongoClients.create() // connects to localhost by default
+val rawCollection = client.getDatabase("test").getCollection("myEntity")
+
+val collection: TypedMongoCollection[MyEntity] = new TypedMongoCollection(rawCollection)
+
+def createEntity(i: Int): MyEntity =
+  MyEntity(s"id$i", i, s"str$i", Some(s"optstr$i"), (i to 10).toList, Map.empty, MyData(i.toDouble, flag = true))
+
+val program: Task[Unit] = for {
+  _ <- collection.insertMany((0 until 10).map(createEntity))
+  _ <- collection.updateMany(
+    MyEntity.ref(_.int) > 5, 
+    MyEntity.ref(_.int).inc(10) && MyEntity.ref(_.str).set("modified")
+  )
+  modifiedEntities <- collection.find(MyEntity.ref(_.str).is("modified")).toListL
+} yield {
+  println(s"Found entities: $modifiedEntities")
+}
+
+import monix.execution.Scheduler.Implicits.global
+program.runToFuture
+```
+
+For more examples of database operations with `TypedMongoCollection`, see
+[tests](https://github.com/AVSystem/scala-commons/blob/mongo-api/commons-mongo/src/test/scala/com/avsystem/commons/mongo/typed/TypedMongoCollectionTest.scala).
+
+`TypedMongoCollection` exposes mostly the same operations as Reactive Streams `MongoCollection` but typed differently, 
```

Comment:
This should be mentioned much earlier in these docs, after all the library doesn't make much sense without it :P